podcasts/Ratings/RatePodcastView.swift

```diff
@@ -1,23 +1,154 @@
+                .buttonize {
+                    viewModel.dismiss()
+                }
+        }
+        .alert(isPresented: $viewModel.anErrorOccurred) {
+            errorAlert
         }
         .frame(maxWidth: .infinity)
+        .padding()
         .applyDefaultThemeOptions()
     }
+
+    @ViewBuilder
+    private var content: some View {
+        switch viewModel.userCanRate {
+        case .checking:
+            ProgressView()
+                .tint(theme.primaryText01)
+                .controlSize(.large)
+        case .allowed:
+            rate
+        case .disallowed:
+            cannotRate
+        }
+    }
+
+    private var cannotRate: some View {
+        Group {
+            PodcastCover(podcastUuid: viewModel.podcast.uuid, big: true)
+                .frame(width: 164, height: 164)
+                .padding(.bottom, 40)
+            Text(L10n.ratingListenToThisPodcastTitle)
+                .font(size: 20, style: .title3, weight: .bold)
+                .padding(.bottom, 16)
+            Text(L10n.ratingListenToThisPodcastMessage)
+                .font(style: .body)
+                .multilineTextAlignment(.center)
+        }
+    }
+
+    private var rate: some View {
+        Group {
+            PodcastCover(podcastUuid: viewModel.podcast.uuid, big: true)
+                .frame(width: 164, height: 164)
+                .padding(.bottom, 40)
+            Text(L10n.ratingTitle(viewModel.podcast.title ?? ""))
+                .font(size: 20, style: .title3, weight: .bold)
+                .padding(.bottom, 16)
+                .multilineTextAlignment(.center)
+            ContentSizeGeometryReader { reader in
+                HStack {
+                    ForEach(0..<Constants.maxStars, id: \.self) { index in
+                        let currentStar = viewModel.stars - Double(index)
+
+                        Group {
+                            if currentStar > 0 && currentStar < 1 {
+                                Image("star-half")
+                                    .resizable()
+                                    .renderingMode(.template)
+                            } else if currentStar > 0 {
+                                Image("star-full")
+                                    .resizable()
+                                    .renderingMode(.template)
+                            } else {
+                                Image("star")
+                                    .resizable()
+                                    .renderingMode(.template)
+                            }
+                        }
+                        .foregroundStyle(theme.primaryText01)
+                        .frame(width: 36, height: 36)
+                        .padding(4)
+                        .onTapGesture {
+                            viewModel.stars = Double(index) + 1
+                        }
+                    }
+                }
+                .gesture(
```

Comment:
When dragging there was a few times where the view started dismissing. using a `highPriorityGesture` prevents that.

```suggestion
                .highPriorityGesture(
```

Reply:
Done in 99c58549dd40a70c7fab99037b82cf5369b663c3